### PR TITLE
Bump version to 4.2.1-SNAPSHOT

### PR DESCRIPTION
**Pull request by sreevatsanraman**
Build passed: 

```[INFO] CDAP KMS .......................................... SUCCESS [0.536s]
[INFO] CDAP Master ....................................... SUCCESS [1.867s]
[INFO] CDAP Operational Stats Core ....................... SUCCESS [1.258s]
[INFO] CDAP Distributions ................................ SUCCESS [0.445s]
[INFO] CDAP Application Archetypes ....................... SUCCESS [0.328s]
[INFO] CDAP Spark Java Archetype ......................... SUCCESS [0.531s]
[INFO] CDAP Spark Scala Archetype ........................ SUCCESS [0.542s]
[INFO] CDAP MapReduce Archetype .......................... SUCCESS [0.362s]
[INFO] CDAP Application Archetype ........................ SUCCESS [0.336s]
[INFO] CDAP Integration Test Framework ................... SUCCESS [1.027s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 4:11.247s
[INFO] Finished at: Tue Jun 13 15:53:38 PDT 2017
[INFO] Final Memory: 296M/754M
```

### CODE REVIEW

File: cdap-distributions/src/hdinsight/pkg/mainTemplate.json [hunk 1]
```diff
@@ -50,12 +50,12 @@
       },
       "installScriptActions": [{
         "name": "[concat('cdap-pageblob-configure-v0','-' ,uniquestring(variables('applicationName')))]",
-        "uri": "http://downloads.cask.co/hdinsight/pageblob-configure-4.2.0.sh",
+        "uri": "http://downloads.cask.co/hdinsight/pageblob-configure-4.2.1-SNAPSHOT.sh",
```
Comment (albertshau) [inline cdap-distributions/src/hdinsight/pkg/mainTemplate.json]: will this cause any problems?

**Comment by albertshau** [comment]
just one question about hdinsights, otherwise lgtm.

File: cdap-distributions/src/hdinsight/pkg/mainTemplate.json (hunk 1 above)
Comment (sreevatsanraman) [reply cdap-distributions/src/hdinsight/pkg/mainTemplate.json]: Got feedback from @dereklwood - changing it based on his reco.

Comment (dereklwood) [reply cdap-distributions/src/hdinsight/pkg/mainTemplate.json]: snapshot wont cause any problems, but next release version is best.

**Comment by dereklwood** [comment]
i think you left out ``cdap-distributions/src/[emr, hdinsight]/install.sh`` -> ``4.2.1-1``

**Comment by sreevatsanraman** [comment]
@dereklwood updated.

**Comment by dereklwood** [comment]
👍 lgtm

**Pull request merged**